ui/src/features/Canvas/components/NodeContextMenu/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { Eye, GearFine, Graph, Trash } from "@phosphor-icons/react";
+import { useCallback } from "react";
+
+import { ContextMenu, ContextMenuMeta } from "@flow/components";
+import { useEditorContext } from "@flow/features/Editor/editorContext";
+import { useT } from "@flow/lib/i18n";
+import { isActionNodeType, Node } from "@flow/types";
+
+type Props = {
+  node: Node;
+  contextMenu: ContextMenuMeta;
+  onClose: () => void;
+};
+
+const NodeContextMenu: React.FC<Props> = ({ node, contextMenu, onClose }) => {
+  const t = useT();
+  const { onNodesChange, onSecondaryNodeAction } = useEditorContext();
+  const { id } = node;
+  const handleNodeDelete = useCallback(() => {
+    onNodesChange?.([{ id, type: "remove" }]);
+  }, [id, onNodesChange]);
+
+  const handleSecondaryNodeAction = useCallback(() => {
+    if (!id) return;
+    onSecondaryNodeAction?.(undefined, id, node.data.subworkflowId);
+  }, [id, node.data.subworkflowId, onSecondaryNodeAction]);
+
+  const menuItems = [
+    node.type === "subworkflow"
+      ? {
+          label: t("Open Subworkflow Canvas"),
+          icon: <Graph weight="light" />,
+          onCallback: handleSecondaryNodeAction,
+          onClose,
```

Comment:
Instead of passing onClose for each one, and inside the ContextMenuItem calling BOTH onCallback and onClose, just combine them in the onCallback.
Also, memoize menuItems 🙏🏻